src/exportJarSteps/ExportJarTaskProvider.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import {
+    CustomExecution, Event, EventEmitter, Pseudoterminal, Task, TaskDefinition,
+    TaskProvider, TaskRevealKind, TaskScope, TerminalDimensions,
+} from "vscode";
+import { createJarFile } from "../exportJarFileCommand";
+import { getExportJarTargetPath } from "../settings";
+import { IStepMetadata } from "./IStepMetadata";
+
+export class ExportJarTaskProvider implements TaskProvider {
+
+    public static exportJarType: string = "exportjar";
+
+    public static getTask(stepMetadata: IStepMetadata): Task {
+        const targetPathSetting: string = getExportJarTargetPath();
+        const defaultDefinition: IExportJarTaskDefinition = {
+            type: ExportJarTaskProvider.exportJarType,
+            targetPath: targetPathSetting,
+            elements: [],
+            mainMethod: undefined,
+        };
+        const task: Task = new Task(defaultDefinition, TaskScope.Workspace, "DEFAULT_EXPORT", ExportJarTaskProvider.exportJarType,
+            new CustomExecution(async (resolvedDefinition: TaskDefinition): Promise<Pseudoterminal> => {
```

Comment:
is `async` needed here?

Reply:
Yes. `CustomExecution` needs a callback function, which returns `Thenable<Pseudoterminal>`, as its parameter.